0015-estimator-interface.md

```diff
@@ -0,0 +1,500 @@
+
+```python
+@dataclass
+class BaseTask:
+    circuit: QuantumCircuit
+```
+
+This base class can be added to Qiskit if and when needed as a non-breaking change, and has been included here mainly to make the concepts clear, rather than to indicate that we need such a base class immediately.
+Most relevant, for the `Estimator` primitive, in order to satisfy the definition as stated above, we propose the task structure
+
+```python
+class EstimatorTask(BaseTask):
+    observables: ObservablesArray = None
+    parameter_values: BindingsArray = None, 
+```
+
+We expect the formal primitive API and primitive implementations to have a strong sense of tasks, but we will not demand that users construct them manually in Python as they are little more than simple data containers (incidentically implemented as dataclasses above), and we do not wish to overburden them with types. This is discussed further in [Type Coercion Strategy](#type-coercion-strategy).
+
+### BindingsArray <a name="bindingsarray"></a>
+
+It is common for a user to want to do a sweep over parameter values, that is, to execute the same parametric circuit with many different parameter binding sets. 
+We propose a new class `BindingsArray` that stores multiple sets of parameters, which can be bound to a circuit. 
+For example, if a circuit has 200 parameters, and a user wishes to execute the circuit for 50 different sets of values, then a single instance of `BindingsArray` could represent 50 sets of 200 parameter values. 
+Moreover, we propose that it be array-like, so that, in this example, the `BindingsArray` instance would be one-dimensional and have shape equal `(50,)`. 
+
+The proposed `BindingsArray` object would support, at a minimum, the following constructor examples for a circuit with three input parameters `a`, `b`, and `c`, where we will use `<>` to denote some `array_like` of the specified shape:
+
+```python
+# a single value for each a, b, and c; 0-dimensional, and compatible with current interface
+BindingsArray([0.0, 1.0, 2.0])
+
+# a single value for each a, b, and c; 0-dimensional, and compatible with current interface
+# satisfies user wish to be able to specify values by name
+BindingsArray(kwvals={Parameter("a"): 0.0, Parameter("b"): 1.0, Parameter("c"): 2.0})
+
+# specify all 50 binding parameter sets in one big array
+BindingsArray(<50, 3>) 
+
+# specify bindings separately for each parameter, required if they have different types
+BindingsArray([<50>, <50>, <50>]) 
+
+# include parameter names with the arrays, where parameters can be grouped together in tuples, or supplied separately
+BindingsArray(kwargs={(a, c): <50, 2>, b: <50>}) 
+
+# “args” and “kwargs” can be mixed
+BindingsArray(<50, 2>, {c: <50>}) 
```

Comment:
Not sure what the identifiers `a`, `b`, and `c` are.  Perhaps you mean the following?

```suggestion
BindingsArray(kwargs={(Parameter("a"), Parameter("c")): <50, 2>, Parameter("b"): <50>}) 

# “args” and “kwargs” can be mixed
BindingsArray(<50, 2>, {Parameter("c"): <50>}) 
```